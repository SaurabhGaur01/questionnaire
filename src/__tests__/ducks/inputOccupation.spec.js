import {
    setOccupation,
    clearOccupation,
    inputOccupationReducer,
} from '../../ducks/userInput/inputOccupation';

describe('inputOccupation Duck Tests', () => {
    const mockObject = "mockValue";

    const unknownAction = {
        type: 'UNKNOWN',
    }

    describe('Action Generator', () => {
        it('setOccupation should match the snapshot' , () => {
            expect(setOccupation(mockObject)).toMatchSnapshot();
        });
        it('clearOccupation should match the snapshot' , () => {
            expect(clearOccupation()).toMatchSnapshot();
        });
    });

    describe('Reducer', () => {
        it('should return the initial state if called with undefined' , () => {
            expect(inputOccupationReducer()).toEqual("");
        });
        it('should return the passed state if called with no action' , () => {
            expect(inputOccupationReducer(mockObject)).toEqual(mockObject);
        });
        it('should return the passed state if called with an unknown action' , () => {
            expect(inputOccupationReducer(mockObject, unknownAction)).toEqual(mockObject);
        });
        it('should return the value which is passed' , () => {
            expect(inputOccupationReducer('', setOccupation(mockObject))).toEqual(mockObject);
        });
        it('should clear the store when called with clearOccupation' , () => {
            expect(inputOccupationReducer('', clearOccupation())).toEqual("");
        });
    });
});