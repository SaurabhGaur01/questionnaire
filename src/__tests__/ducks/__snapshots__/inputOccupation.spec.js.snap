// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`inputOccupation Duck Tests Action Generator clearOccupation should match the snapshot 1`] = `
Object {
  "type": "modules/questionnaire-app/CLEAR_OCCUPATION",
}
`;

exports[`inputOccupation Duck Tests Action Generator setOccupation should match the snapshot 1`] = `
Object {
  "type": "modules/questionnaire-app/SET_OCCUPATION",
  "value": "mockValue",
}
`;
